### ISSUE TITLE
Go envのreadme

### ISSUE THREAD

**Issue by OkanoShinri** [sotetsuk/pgx#55]
## State
```py
@struct.dataclass
class GoState:
    # 横幅, マスの数ではない
    size: jnp.ndarray = jnp.full(1, 19, dtype=int)

    # 連
    ren_id_board: jnp.ndarray = jnp.full((2, 19 * 19), -1, dtype=int)

    # 連idが使えるか
    available_ren_id: jnp.ndarray = jnp.ones((2, 19 * 19), dtype=bool)

    # 連周りの情報 0:None 1:呼吸点 2:石
    liberty: jnp.ndarray = jnp.zeros((2, 19 * 19, 19 * 19), dtype=int)

    # 隣接している敵の連id
    adj_ren_id: jnp.ndarray = jnp.zeros((2, 19 * 19, 19 * 19), dtype=bool)

    # 経過ターン, 0始まり
    turn: jnp.ndarray = jnp.zeros(1, dtype=int)

    # [0]: 黒の得たアゲハマ, [1]: 白の得たアゲハマ
    agehama: jnp.ndarray = jnp.zeros(2, dtype=int)

    # 直前のactionがパスだとTrue
    passed: jnp.ndarray = jnp.zeros(1, dtype=bool)

    # コウによる着手禁止点(xy), 無ければ(-1)
    kou: jnp.ndarray = jnp.full(1, -1, dtype=int)
```

## 関数

- `init(size: int)`
  - `GoState`を返す
  - `size`で横幅を指定する
- `step(state: GoState, action: int, size: int)`
  - `action`は盤面を左上から右に数えていったマス目の番号（整数値）
  - passの時は`action`に`-1`を指定する
  - jaxの都合で盤の`size`を指定する必要あり
  - 返り値は`Tuple[GoState, jnp.ndarray, bool]`で、それぞれ(state, reward, done)
- `legal_actions(state: GoState, size: int)`
  - 長さ`size*size`の`ndarray`を返す。石を置けるマスは`True`、そうでないマスは`False`が入っている
- `show(state: GoState)`
  - 可視化する
